### ISSUE TITLE
Note to self: Add an LLM example

### ISSUE THREAD

**Issue by rasbt** [Raschka-research-group/coral_pytorch#29]

**Comment by dleve123** [comment]
This issue perhaps seems addressed by https://github.com/Raschka-research-group/coral-pytorch/pull/30?

On a related note, I notice that you only report MAE in the LLM examples - considering most possible users of this method will be evaluating relative to a multi-class classification problem framing (at least I am 😄) , I think it would be really useful to show the impact of CORN on test accuracy relative to a CE baseline. As such, did you save the weights of the models trained in the [CORN notebook](https://github.com/Raschka-research-group/coral-pytorch/blob/main/docs/tutorials/pytorch_lightning/distilbert-corn-tripadvisor.ipynb) and the [CE notebook](https://github.com/Raschka-research-group/coral-pytorch/blob/main/docs/tutorials/pytorch_lightning/distilbert-crossentropy-tripadvisor.ipynb)? If so, I believe it would be pretty straightforward to extend the lightning module with an accuracy metric (I would be happy to help there as well)!

Thanks!

**Comment by rasbt** [comment]
Thanks for the suggestion. Originally, I didn't want to add that because I think this would be misleading: accuracy on an ordinal problem doesn't make that much sense imho. For multiclass classification, I'd also expect cross entropy to expect best because it basically rewards exact matches and all mistakes are equally wrong.

**Comment by rasbt** [comment]
The original LLM topic could also be closed since it's already addressed 😅
https://github.com/Raschka-research-group/coral-pytorch/blob/main/docs/tutorials/pytorch_lightning/distilbert-corn-tripadvisor.ipynb

**Comment by dleve123** [comment]
> The original LLM topic could also be closed since it's already addressed 😅
https://github.com/Raschka-research-group/coral-pytorch/blob/main/docs/tutorials/pytorch_lightning/distilbert-corn-tripadvisor.ipynb

👍 

> For multiclass classification, I'd also expect cross entropy to expect best because it basically rewards exact matches and all mistakes are equally wrong.

Hmm - not really following you here. Are you suggesting that CORN should only when used when the problem in question is really ordinal – even if model/training approach uses a cross-entropy loss. For instance, it's always better for tripadvisor rating predictions to be closer than farther to their ground truth ordinal value; it just so happens to be easy (before this method!) to model this ordinal regression problem as a multi-class classification problem? Is that right? Can you elaborate a bit here?

**Comment by rasbt** [comment]
Good point, what I wrote sounds indeed confusing. What I meant is that I expect cross entropy to be always better when looking at accuracy. That's because it's optimizing for exact matches (and accuracy only looks for exact matches).

E.g., say you have true labels [1, 7, 8, 9]. 

A cross-entropy classifier may return [9, 9, 9, 9]. That's 25% accuracy. 

A ordinal model may predict [2, 6, 7, 8] -- it's very close but notice that there is no exact match. Here, the accuracy is 0% even though it's a really good prediction that is only slightly off. 

So I am not saying that ordinal regression gets 0%. It can get a decent high accuracy percentage. But I think that looking at accuracy could easily be misinterpreted if someone is not familiar with ordinal regression. So I would prefer not to include it in the evaluation.

PS: Closing this because the original task (having an LLM example) was implemented in a PR. But please feel free to continue the accuracy discussion if you have follow-up comments or questions.
